python/ray/tests/test_minimal_install.py

```diff
@@ -29,14 +34,45 @@ def test_correct_python_version():
     )
 
 
+class MockBaseModel:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __init_subclass__(self, *args, **kwargs):
+        pass
+
+
+def _make_mock_pydantic_modules(pydantic_version: str) -> Dict:
+    """Make a mock for the `pydantic` module.
+
+    This module requires special handling to:
+        - Make `BaseModel` a class object so type hints work.
+        - Set the `__version__` attribute appropriately.
+        - Also mock `pydantic.v1` for `pydantic >= 2.0`.
+        - Also mock `pydantic.dataclasses`.
+
+    Returns a dict of mocked modules.
+    """
+    mock_modules = {
+        "pydantic": mock.MagicMock(),
+        "pydantic.dataclasses": mock.MagicMock(),
+    }
+    mock_modules["pydantic"].BaseModel = MockBaseModel
+    if packaging.version.parse(pydantic_version) >= packaging.version.parse("1.9.0"):
+        mock_modules["pydantic"].__version__ = pydantic_version
+
+    if packaging.version.parse(pydantic_version) >= packaging.version.parse("2.0.0"):
+        mock_modules["pydantic.v1"] = mock_modules["pydantic"]
+
+    return mock_modules
+
+
+@pytest.mark.parametrize("pydantic_version", ["1.8.0", "1.10.0", "2.0.0"])
```

Comment:
Non-blocker: I think in general it would be nice to check the boundary case at version 1.9.0 instead of 1.10.0, but not a blocker as we both knows they should behave the same 😛.

Reply:
updated